swift_browser_ui_frontend/src/components/UploadModal.vue

```diff
@@ -0,0 +1,712 @@
+<template>
+  <c-card class="upload-card">
+    <div class="modal-content-wrapper">
+      <h3 class="title is-3 has-text-dark">
+        {{ $t("message.encrypt.uploadFiles") }}
+      </h3>
+      <c-card-content>
+        <h6 class="title is-6 has-text-dark">
+          1. {{ $t("message.encrypt.upload_step1") }}
+        </h6>
+        <p class="info-text is-size-6">
+          {{ $t("message.container_ops.norename") }}
+        </p>
+        <c-autocomplete
+          v-control
+          v-csc-model="selectedFolder"
+          :items.prop="filteredItems"
+          :query="inputFolder"
+          :label="$t('message.container_ops.folderName')"
+          hide-details
+          required
+          @changeQuery="onQueryChange"
+        />
+        <h6 class="title is-6 has-text-dark">
+          2. {{ $t("message.encrypt.upload_step2") }}
+        </h6>
+        <div
+          class="dropArea is-flex
+                is-align-items-center is-justify-content-center"
+          @dragover="dragHandler"
+          @dragleave="dragLeaveHandler"
+          @drop="navUpload"
+        >
+          <span>{{ $t("message.dropFiles") }}</span>
+          <b-upload
+            v-model="files"
+            multiple
+            accept
+            class="file is-primary"
+          >
+            <span class="file-cta">
+              {{ $t("message.encrypt.dropMsg") }}
+            </span>
+          </b-upload>
+        </div>
+        <c-data-table
+          v-if="dropFiles.length > 0"
+          class="files-table"
+          :data.prop="dropFiles"
+          :headers.prop="fileHeaders"
+          :no-data-text="$t('message.encrypt.empty')"
+          :pagination.prop="filesPagination"
+          :footer-options.prop="footer"
+        />
+        <p class="info-text is-size-6">
+          {{ $t("message.container_ops.createdFolder") }}
+          <b>{{ $t("message.container_ops.myResearchProject") }}</b>
+        </p>
+        <c-link
+          :href="`https://my.csc.fi/myProjects/project/${currentProjectID}`"
```

Comment:
Same as in container creation modal, use `$store.state.active.name` without the `project_` part to make the link work.